### PR TITLE
topology1: add support for HP Omen 16

### PR DESCRIPTION
**Pull request by plbossart**
BugLink: https://github.com/thesofproject/linux/issues/3727

### CODE REVIEW

**Comment by marc-hb** [comment]
2 devices unavailable in https://sof-ci.01.org/sofpr/PR5974/build788/devicetest/ because a temporary lab issue. Everything else green

**Comment by marc-hb** [comment]
SOFCI TEST

**Review by ranj063 — APPROVED** [review]

**Review by RanderWang — APPROVED** [review]

**Comment by plbossart** [comment]
> @plbossart CI is good but I assume we are pending on the PR being tested on the device ?

yep, something is not working apparently see https://github.com/thesofproject/linux/issues/3727

````
[    5.782321] sof-audio-pci-intel-tgl 0000:00:1f.3: tplg: ready widget id 16 pipe 3 type 27 name : ALH770.OUT stream SDW1-Playback
[    5.782324] sof-audio-pci-intel-tgl 0000:00:1f.3: error: can't connect DAI ALH770.OUT stream SDW1-Playback
[    5.782326] sof-audio-pci-intel-tgl 0000:00:1f.3: error: failed to add widget id 0 type 27 name : ALH770.OUT stream SDW1-Playback
````

**Comment by lgirdwood** [comment]
Some failures on ADL xtos - rerun to be sure.

File: tools/topology/topology1/sof-tgl-rt711-rt1308.m4 [hunk 1]
```diff
@@ -224,7 +224,7 @@ ifdef(`EXT_AMP',
 # playback DAI is ALH(SDW1 PIN2) using 2 periods
```
Comment (RanderWang) [inline tools/topology/topology1/sof-tgl-rt711-rt1308.m4:220]: this comments is out of date

**Review by bardliao — APPROVED** [review]

File: tools/topology/topology1/sof-tgl-rt711-rt1308.m4 (hunk 1 above)
Comment (plbossart) [reply tools/topology/topology1/sof-tgl-rt711-rt1308.m4:220]: updated

File: tools/topology/topology1/CMakeLists.txt [hunk 2]
```diff
@@ -129,7 +129,7 @@ set(TPLGS
 	"sof-tgl-rt711-rt1308\;sof-tgl-rt711-4ch\;-DCHANNELS=4\;-DDMICPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_16khz.m4\;-DPLATFORM=tgl"
 	"sof-tgl-rt711-rt1308\;sof-adl-rt711-4ch\;-DCHANNELS=4\;-DBT_OFFLOAD\;-DDMICPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_16khz.m4\;-DPLATFORM=adl"
 	"sof-tgl-rt711-rt1308\;sof-adl-rt711\;-DCHANNELS=0\;-DBT_OFFLOAD\;-DPLATFORM=adl"
-	"sof-tgl-rt711-rt1308\;sof-adl-rt711-l0-rt1316-l3-2ch\;-DCHANNELS=2\;-DEXT_AMP\;-DAMP_1_LINK=3\;-DDMICPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_16khz.m4\;-DPLATFORM=adl"
+	"sof-tgl-rt711-rt1308\;sof-adl-rt711-l0-rt1316-l3-2ch\;-DCHANNELS=2\;-DEXT_AMP\;-DEXT_AMP_REF\;-DAMP_1_LINK=3\;-DDMICPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_48khz.m4\;-DDMIC16KPROC_FILTER1=eq_iir_coef_highpass_40hz_20db_16khz.m4\;-DPLATFORM=adl"
```
[the remark this answers is not in the document]
Comment (plbossart) [reply tools/topology/topology1/CMakeLists.txt:134]: I wanted to keep this separate to make sure we keep the rt1316 case in mind.

**Review by aiChaoSONG — APPROVED** [review]
only a single unrelated timeout.

**Pull request merged**